### PR TITLE
TSK-1197: readded test case for testing the ordering of the databases…

### PR DESCRIPTION
**Pull request by sfrevel**
TSK-1197: readded test case for testing the ordering of the databases H2, Postgress and DB2; remove Db2 10.5 travis job

TSK-1197: build Postgres image with german collation

### CODE REVIEW

File: ci/prepare_db.bat [hunk 1]
```diff
@@ -107,13 +107,17 @@ SET PROP_FILE=%HOMEPATH%\taskanaUnitTest.properties
...
+        docker-compose -f %~dp0/docker-compose.yml start taskana-postgres_10_4
     )
 
     IF NOT DEFINED CONTAINER_EXISTS (
-        ECHO docker run -d -p 50102:5432 --name taskana-postgres_10_4 -e POSTGRES_PASSWORD=postgres postgres:10.4
-        docker run -d -p 50102:5432 --name taskana-postgres_10_4 -e POSTGRES_PASSWORD=postgres postgres:10.4
+        REM ECHO docker run -d -p 50102:5432 --name taskana-postgres_10_4 -e POSTGRES_PASSWORD=postgres postgres:10.4
+        REM docker run -d -p 50102:5432 --name taskana-postgres_10_4 -e POSTGRES_PASSWORD=postgres postgres:10.4
+        ECHO docker-compose -f %~dp0/docker-compose.yml up -d
+        docker-compose -f %~dp0/docker-compose.yml up -d
```
Comment (mustaphazorgati) [inline ci/prepare_db.bat:120]: there is no need for this seperation anymore. `docker-compose up -d` recreates the container if already existing.

File: ci/prepare_db.bat [hunk 2]
```diff
@@ -107,13 +107,17 @@ SET PROP_FILE=%HOMEPATH%\taskanaUnitTest.properties
...
     )
 
     ECHO jdbcDriver=org.postgresql.Driver> %PROP_FILE%
```
Comment (mustaphazorgati) [inline ci/prepare_db.bat:123]: I just started a polly.. Maybe this file is not needed anymore. Let's wait for the feedback in #taskana.

File: ci/prepare_db.sh [hunk 3]
```diff
@@ -63,9 +63,9 @@ function main() {
   POSTGRES_10_4)
```
Comment (mustaphazorgati) [inline ci/prepare_db.sh:63]: please rename this to POSTGRES_10

File: ci/prepare_db.sh [hunk 4]
```diff
@@ -63,9 +63,9 @@ function main() {
...
     if [[ -z $(docker ps -aq -f name=^/taskana-postgres_10_4$ -f status=running) ]]; then
       if [[ -z $(docker ps -aq -f name=^/taskana-postgres_10_4$) ]]; then
-        docker run -d -p 50102:5432 --name taskana-postgres_10_4 -e POSTGRES_PASSWORD=postgres postgres:10.4
+		 docker-compose -f `dirname $0`/docker-compose.yml up -d
       else
-        docker start taskana-postgres_10_4
+		docker-compose -f `dirname $0`/docker-compose.yml start taskana-postgres_10_4
       fi
     fi
```
Comment (mustaphazorgati) [inline ci/prepare_db.sh:70]: no need to differentiate anymore. `docker-compose up -d` is enough. See comment above

File: ci/prepare_db.sh [hunk 5]
```diff
@@ -63,9 +63,9 @@ function main() {
   POSTGRES_10_4)
     if [[ -z $(docker ps -aq -f name=^/taskana-postgres_10_4$ -f status=running) ]]; then
       if [[ -z $(docker ps -aq -f name=^/taskana-postgres_10_4$) ]]; then
-        docker run -d -p 50102:5432 --name taskana-postgres_10_4 -e POSTGRES_PASSWORD=postgres postgres:10.4
+		 docker-compose -f `dirname $0`/docker-compose.yml up -d
       else
-        docker start taskana-postgres_10_4
```
Comment (mustaphazorgati) [inline ci/prepare_db.sh:68]: I am a big fan of making an extra variable for `dirname $0`. But you don't have to ;)

File: lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskServiceImpl.java [hunk 6]
```diff
@@ -261,7 +261,7 @@ public Task createTask(Task taskToCreate)
         // primary key violation: "UC_EXTERNAL_ID_INDEX_2 ON TASKANA.TASK(EXTERNAL_ID) ...
         String msg = e.getMessage() != null ? e.getMessage().toLowerCase() : null;
         if (msg != null
-            && (msg.contains("violation") || msg.contains("violates"))
+            && (msg.contains("violation") || msg.contains("violates") || msg.contains("verletzt"))
```
Comment (mustaphazorgati) [inline lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskServiceImpl.java:264]: This looks like a big code smell due to the dependency to the database locale.
Ticket?

Comment (sfrevel) [reply lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskServiceImpl.java:264]: Yes I also think a ticket would be great. Maybe there is a way to get this done in a better way. I always thought that all DBs have an explicit "UniqueContraintViolation" exception. It seems that is not the case and therefore this workaround is implemented.

File: ci/prepare_db.bat (hunk 2 above)
Comment (sfrevel) [reply ci/prepare_db.bat:123]: 👍

**Comment by mustaphazorgati** [comment]
General question: Did you think about modifying the documentation in taskana.pro ?

**Comment by sfrevel** [comment]
Docu changes on the following pages:
- https://taskana.atlassian.net/wiki/spaces/TAS/pages/184254465/How+to+install+and+use+DB2+locally
- https://taskana.atlassian.net/wiki/spaces/TAS/pages/784793601/Travis+configuration
- https://taskana.atlassian.net/wiki/spaces/TAS/pages/698286081/Using+multiple+databases+for+taskana-core+unit+tests

File: lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskServiceImpl.java (hunk 6 above)
Comment (mustaphazorgati) [reply lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskServiceImpl.java:264]: did you create the ticket?

**Comment by mustaphazorgati** [comment]
> Docu changes on the following pages:
> 
> * https://taskana.atlassian.net/wiki/spaces/TAS/pages/184254465/How+to+install+and+use+DB2+locally
> * https://taskana.atlassian.net/wiki/spaces/TAS/pages/784793601/Travis+configuration
> * https://taskana.atlassian.net/wiki/spaces/TAS/pages/698286081/Using+multiple+databases+for+taskana-core+unit+tests

thanks for the documentation. I've removed some additional references for db2 10.5 and corrected some minor things. If you're interested I can let you know by tomorrow what I did exactly (I can't tell you which pages I edited ... I'll see it in the daily report tonight xD)

**Comment by sfrevel** [comment]
I checked you editings in the docu via confluence status mails of the pages and again I forgot the DB2_10_5 ;)

File: lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskServiceImpl.java (hunk 6 above)
Comment (sfrevel) [reply lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskServiceImpl.java:264]: Yes, now i did ;D
https://taskana.atlassian.net/browse/TSK-1231

**Pull request merged**